gensrc/thrift/Types.thrift

```diff
@@ -397,7 +397,8 @@ enum TTableType {
     FILE_TABLE,
     DELTALAKE_TABLE,
     TABLE_FUNCTION_TABLE,
-    ODPS_TABLE
+    ODPS_TABLE,
+    LOGICAL_ICEBERG_METADATA_TABLE
```

Comment:
do we need to have different metadata table like `LOGICAL_HUDI_METADATA_TABLE` or `LOGICAL_DELTA_LAKE_METADATA_TABLE` in the future?